### PR TITLE
Add lookup feature

### PR DESCRIPTION
**Pull request by jaymo001**

### CODE REVIEW

File: feathr_project/feathr/aggregation.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from enum import Enum
+
+class Aggregation(Enum):
+    NOP = 0
```
Comment (hangfei) [inline feathr_project/feathr/aggregation.py:4]: explain what does each aggregation do

File: feathr_project/feathr/dtype.py [hunk 2]
```diff
@@ -126,4 +149,6 @@ def __init__(self, shape: int, dType: ValueType = ValueType.INT32):
 DOUBLE = DoubleFeatureType()
 STRING = StringFeatureType()
 BYTES = BytesFeatureType()
-FLOAT_VECTOR = FloatVectorFeatureType()
\ No newline at end of file
+FLOAT_VECTOR = FloatVectorFeatureType()
+INT32_VECTOR = Int32VectorFeatureType()
```
Comment (hangfei) [inline feathr_project/feathr/dtype.py:153]: add INT64_vector?

File: feathr_project/feathr/dtype.py [hunk 3]
```diff
@@ -126,4 +149,6 @@ def __init__(self, shape: int, dType: ValueType = ValueType.INT32):
...
+DOUBLE_VECTOR = DoubleVectorFeatureType()
```
Comment (hangfei) [inline feathr_project/feathr/dtype.py:154]: add sparse 1d array?
we support int to string, float, double, boolean sparse array. (but can be next PR)

File: feathr_project/feathr/lookup_feature.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+from feathr.dtype import FeatureType
+from feathr.feature import FeatureBase
+from feathr.transformation import RowTransformation
+from feathr.typed_key import DUMMY_KEY, TypedKey
+from feathr.aggregation import Aggregation
+
+class LookupFeature(FeatureBase):
+    """A lookup feature is a feature defined on top of two other features, i.e. using the feature value 
+    of the base feature as key, to lookup the feature value from the expansion feature.
```
Comment (hangfei) [inline feathr_project/feathr/lookup_feature.py:14]: this is good description but due to the complexity of it, it's better to provide an example here.

File: feathr_project/test/test_lookup_feature.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
+from feathr.feature import Feature
+from feathr.lookup_feature import LookupFeature
+from feathr.dtype import FLOAT, FLOAT_VECTOR, ValueType, INT32_VECTOR
+from feathr.typed_key import TypedKey
```
Comment (hangfei) [inline feathr_project/test/test_lookup_feature.py:4]: use the simplified from feathr import ...

File: feathr_project/feathr/lookup_feature.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    Attributes:
+        name: Derived feature name
+        feature_type: Type of derived feature
+        key: Join key of the derived feature
+        base_feature: The feature value of this feature will be used as key to lookup from the expansion feature
+        expansion_feature: The feature to be lookup
```
Comment (d4ve) [inline feathr_project/feathr/lookup_feature.py:20]: s/lookup/looked up/

**Review by d4ve — COMMENTED** [review]
generally lgtm, other than the issues Hangfei pointed out

**Review by hangfei — APPROVED** [review]

**Review by xiaoyongzhu — APPROVED** [review]

**Pull request merged**